frontend/src/component/personalDashboard/createChartOptions.ts

```diff
@@ -0,0 +1,161 @@
+            borderRadius: 5,
+        },
+    },
+    interaction: {
+        mode: 'index',
+        intersect: false,
+    },
+});
+
+export const createBarChartOptions = (
+    theme: Theme,
+    hoursBack: number,
+    locationSettings: ILocationSettings,
+): ChartOptions<'bar'> => {
+    const { plugins, responsive, elements, interaction, scales } =
+        createPlaceholderBarChartOptions(theme);
+    return {
+        plugins: {
+            legend: plugins?.legend,
+            tooltip: {
+                backgroundColor: theme.palette.background.paper,
+                titleColor: theme.palette.text.primary,
+                bodyColor: theme.palette.text.primary,
+                bodySpacing: 6,
+                padding: {
+                    top: 20,
+                    bottom: 20,
+                    left: 30,
+                    right: 30,
+                },
+                borderColor: 'rgba(0, 0, 0, 0.05)',
+                borderWidth: 3,
+                usePointStyle: true,
+                caretSize: 0,
+                boxPadding: 10,
+                callbacks: {
+                    label: (item) => {
+                        return `${item.formattedValue} - ${item.dataset.label}`;
+                    },
+                    afterLabel: (item) => {
+                        const data = item.dataset.data[
+                            item.dataIndex
+                        ] as unknown as IPoint;
+
+                        if (
+                            item.dataset.label !== 'Exposed' ||
+                            data.variants === undefined
+                        ) {
+                            return '';
+                        }
+                        const { disabled, ...actualVariants } = data.variants;
+                        return Object.entries(actualVariants)
+                            .map((entry) => formatVariantEntry(entry, data.y))
+                            .join('\n');
+                    },
+                    title: (items) => {
+                        return `Time: ${
+                            hoursBack > 48
+                                ? formatDateYMDHM(
+                                      items[0].label,
+                                      locationSettings.locale,
+                                      'UTC',
+                                  )
+                                : formatDateHM(
+                                      items[0].label,
+                                      locationSettings.locale,
+                                  )
+                        }`;
+                    },
+                },
+            },
+        },
+        responsive,
+        scales: {
+            x: {
+                ...(scales ? scales.x : {}),
+                ticks: {
+                    color: theme.palette.text.secondary,
+                    callback(tickValue) {
+                        const label = this.getLabelForValue(Number(tickValue));
```

Comment:
this was tricky to find. chartjs doesn't return timestamps in the callback but indices. So I lookup the actual value using a utility function provided in the latest version of chartjs